src/components/Select/index.jsx

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React, { useEffect, useState, useRef } from 'react';
+
+import LinearGradientContainer from './LinearGradientContainer';
+import styles from './styles.scss';
+import { SelectContext, useSelectContext } from './selectContext';
+
+const classNames = require('classnames');
+
+const useOnClickOutside = (ref, handler) => {
+  useEffect(() => {
+    const listener = (event) => {
+      const el = ref?.current;
+      if (!el || el.contains(event?.target || null)) {
```

Comment:
event?.target || null - no reason
so  prob can be enough:
if (!ref.current || ref.current.contains(event.target))